lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_facts.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018
+# Gustavo Muniz do Carmo <gustavo@esign.com.br>
+# Zim Kalinowski <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_virtualmachine_facts
+
+version_added: "2.7"
+
+short_description: Get virtual machine facts.
+
+description:
+  - Get facts for all virtual machines of a resource group.
+
+options:
+    resource_group:
+        description:
+        - Name of the resource group containing the virtual machines (required when filtering by vm name).
+    name:
+        description:
+        - Name of the virtual machine.
+    tags:
+        description:
+        - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+
+extends_documentation_fragment:
+  - azure
+
+author:
+  - "Gustavo Muniz do Carmo (@gustavomcarmo)"
+  - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Get facts for all virtual machines of a resource group
+    azure_rm_virtualmachine_facts:
+      resource_group: Testing
+
+  - name: Get facts by name
+    azure_rm_virtualmachine_facts:
+      resource_group: Testing
+      name: vm
+
+  - name: Get facts by tags
+    azure_rm_virtualmachine_facts:
+      resource_group: Testing
+      tags:
+        - testing
+        - foo:bar
+'''
+
+RETURN = '''
+azure_virtualmachines:
+    description: List of resource group's virtual machines dicts.
+    returned: always
+    type: list
+    example: [{}]
```

Comment:
I like the new format, the last bit is documenting each returned key and what they represent here. Have a look here for an example of how this is done https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_instance_facts.py#L66